lib/elixir/lib/enum.ex

```diff
@@ -2478,6 +2478,9 @@ defmodule Enum do
       iex> Enum.reduce([1, 2, 3], 0, fn x, acc -> x + acc end)
       6
 
+      iex> Enum.reduce(%{a: 2, b: 3, c: 4}, %{}, fn {key, val}, acc -> Map.put(acc, key, val * 2) end)
+      %{a: 4, b: 6, c: 8}
```

Comment:
I like the idea of adding this example, but I think we should avoid adding an example which would likely be better achieved via `Map.map/2` or `Map.new/2`. Thoughts?

Reply:
🤔 ...or maybe I should put this example on Map.new and start using that 😅 (but Map.map is gone, isn't it? It existed for like 1-2 point releases, then gone. Definitely doesn't show in 1.14 docs)

Reply:
Sorry yes that's gone, but `Map.new/2` is what I'd use here. I still think an additional example here can be nice, just not the one that can be implemented as `Map.new/2` 🙃

Reply:
Ok, how about this new commit, then? Moved that example to `Map.new/2` and changed this one to sum the values